src/python/pants/goal/run_tracker.py

```diff
@@ -414,21 +414,77 @@ def shutdown_worker_pool(self):
     """
     SubprocPool.shutdown(self._aborted)
 
-  def report_target_info(self, scope, target, key, val):
+  @staticmethod
+  def create_dict_with_nested_keys_and_val(all_keys, value, index):
```

Comment:
Since this is an internal helper, not intended to be used outside this class, it's good practice to have the name start with an underscore (`_create_dict_with_nested_keys_and_val`). This is idiomatic python for "this is a 'private' method".